v3/src/models/formula/functions/string-functions.ts

```diff
@@ -92,7 +96,7 @@ export const stringFunctions = {
     evaluate: (...args: FValue[]) => {
       const stringToLookIn = valueToString(args[0]),
         pattern = valueToString(args[1])
-      return (stringToLookIn.match(new RegExp(escapeStringRegexp(pattern), "gmi")) || []).length
+      return (stringToLookIn.match(new RegExp(unescapeBacktickString(pattern), "gmi")) || []).length
```

Comment:
Still need to sanitize the input, so it would have to be something like:
```suggestion
      return (stringToLookIn.match(new RegExp(escapeStringRegexp(unescapeBacktickString(pattern)), "gmi")) || []).length
```

Reply:
Unfortunately, I don't think this will work for the StoryQ plugin. It sends patterns like `\\\\bword\\\\b`, which reach the function as `\\bword\\b`. `unescapeBacktickString` translates this to `\bword\b`, which is required for the function to work properly. I don't think there's any way to call `escapeStringRegexp` on this string without adding extra `\`, breaking the function.

Reply:
We can put this off to later if we don't want to deal with it now, but the right thing to do in that case would be to extract `word`, escape it, and then wrap it in `\b` again. Passing external input directly to a regex is like passing user input directly to SQL -- never a good/safe idea -- just ask [Little Bobby Tables](https://xkcd.com/327/). 😜

Reply:
I can understand that this is important for security, but it's going to be tricky to get it to work with old plugins. v2 feeds patterns directly into RegExps without any processing. Playing around with StoryQ, I've already found one other example of patterns with `\` (`\\\\!` for a punctuation feature), and I wouldn't be surprised if there were more. Will we be able to catch all of these cases with individual special rules?

The alternative would be to change the plugins, which we might want to do. But what could StoryQ pass to Codap to get the `\bword\b` pattern? `escapeStringRegexp` will not add a `\` to `b`, and if StoryQ includes any `\`s, they will become `\\` when we run `escapeStringRegexp` on the pattern.

Reply:
So stick with v2 behavior for now and add a TODO to return to it at some point.

Reply:
I was thinking about this more recently and another approach would be to do the regex matching in a WebWorker which could be canceled if it took too long. In any case, we can certainly put this off for another day.